qadence/model.py

```diff
@@ -432,6 +432,30 @@ def _from_dict(cls, d: dict, as_torch: bool = False) -> QuantumModel:
 
         return qm
 
+    def load_params_from_dict(self, d: dict) -> None:
+        """Copy parameters from dictionary into this QuantumModel.
+
+        Unlike :meth:`~qadence.QuantumModel.from_dict`, this method does not create a new
+        QuantumModel instance, but rather loads the parameters into the same QuantumModel.
+        The behaviour of this method is similar to :meth:`~torch.nn.Module.load_state_dict`.
+
+        The dictionary is assumed to have the format as saved via
+        :meth:`~qadence.QuantumModel.to_dict`
+
+        Args:
+            d (dict): Dictionary
+        """
+        param_dict = d["param_dict"]
+        for n, param in param_dict.items():
+            if n in param_dict:
```

Comment:
Indeed thanks, this line can go

Reply:
Just replace by `if n is in self._params` as you try to access it one line below

Reply:
Actually, wouldn't it be better to let the `try`/`except` statement catch the error in the case where the key does not exist? Otherwise it will go silent and the user might think that all the parameters are loaded correctly when in reality they were not

Reply:
A more elegant solution would be to check what are the `missing_keys` and the `unexpected_keys` in the dictionary, as they do in the [`torch.nn.Module.load_state_dict`]( https://github.com/pytorch/pytorch/blob/d95756f6a5a65be0a39bdfa970fe3f9608a19288/torch/nn/modules/module.py#L2460) method in torch

Reply:
Indeed, so either we do the check, or simply catch the KeyError. Timeline is up to you.

Reply:
Great, I think it makes sense to do the check. I'll add it here